src/test/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewControllerTest.java

```diff
@@ -0,0 +1,161 @@
+package org.cbioportal.application.rest.vcolumnstore;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Arrays;
+import java.util.Collections;
+import org.cbioportal.domain.studyview.StudyViewService;
+import org.cbioportal.legacy.web.parameter.GenomicDataCountFilter;
+import org.cbioportal.legacy.web.parameter.StudyViewFilter;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+/**
+ * Unit tests for ColumnarStoreStudyViewController.
+ *
+ * Tests for Issue #11206: 500 error evaluating genomicDataFilter
+ * When genomicDataFilters is empty or null, the endpoints should return
+ * an empty list instead of throwing an IndexOutOfBoundsException.
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class ColumnarStoreStudyViewControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private StudyViewService studyViewService;
+
+    private static final String TEST_STUDY_ID = "test_study";
+
+    /**
+     * Test that fetchGenomicDataCounts returns empty list when genomicDataFilters is empty.
+     * This verifies the fix for Issue #11206.
```

Comment:
we probably don't need this line of comment with issue reference. same applies to the other tests below.